### DISCUSSION TITLE
Help with custom titlecase function?

### DISCUSSION THREAD

**Discussion by trapd00r** [beetbox/beets#3998] [Q&A] [answered]
When using %title{}, python tries to uppercase the first character no matter what it is. This is bad where things like '(instrumental' is considered one word entity. 
I would like to uppercase the first ascii character in a string instead but my python isn't the best. I've made a few things using the inline plugin but this one I can't figure out.

`04 Fight Music (instrumental Version).mp3`

**Comment by DjSlash** [comment]
Python's title() works like you expect:
```python
>>> "hallo doei dag".title()
'Hallo Doei Dag'
>>> "{hallo doei dag".title()
'{Hallo Doei Dag'
>>> "(hallo doei dag".title()
'(Hallo Doei Dag'
```
So, beets' %title parses differently.

**Reply by wisp3rwind** [reply]
The template uses `string.capwords()` instead of `str.title()`. According to https://docs.python.org/3.9/library/stdtypes.html?highlight=str%20title#str.title, the latter isn't perfect either.

**Comment by trapd00r** [comment]
Thank you! Can I create a function similar to this using the inline plugin, that I can use just like %title{$var}, that is, use my function on whatever value I pass to it? So far I'm only aware that I can use an explicit value, like $album, and access it by $titlecase_proper in my path format.

```
item_fields:                                                                                                                                               
                                                                                                                                                            
   titlecase_proper: |                                                                                                                                      
     import re                                                                                                                                              
     def titlecase(s):                                                                                                                                      
       return re.sub(r"[A-Za-z]+('[A-Za-z]+)?",                                                                                                             
                    lambda mo: mo.group(0).capitalize(),                                                                                                   
                     s)                                                                                                                                     
     return s                                                      
```

**Comment by trapd00r** [comment, marked as answer]
I made a [plugin](https://github.com/trapd00r/beets-titlecase_proper).
